### PR TITLE
[unit-test]: overflow menu

### PR DESCRIPTION
**Pull request by divya-281**
increases test coverage for overflow menu
#17499

### CODE REVIEW

**Comment by guidari** [comment]
Hey @divya-281, thanks for opening that PR!! 🚀
Do you think we could raise the coverage to over 80%? Currently it is 67,41%. But if there is no way to tests those parts, don't worry.

https://app.codecov.io/gh/carbon-design-system/carbon/pull/17875/blob/packages/react/src/components/OverflowMenu/OverflowMenu.tsx

**Comment by divya-281** [comment]
@guidari  I was able to increase to 74%

**Review by tay1orjones — COMMENTED** [review]
This is looking good, thanks for taking it on! 

Looking at the coverage report, a few ideas to get this to 80%:

* Test that pressing <kbd>escape</kbd> closes the menu
* Find the path to get `this.closeMenuAndFocus();` to be called. The steps to reproduce [this bug](https://github.com/carbon-design-system/carbon/issues/12942) could be helpful, as [the fix](https://github.com/carbon-design-system/carbon/pull/15375) was within the final condition before it's called. I think the assertion would be that the menu closed and focus returned to the trigger element.

https://github.com/carbon-design-system/carbon/blob/796f838c420d0356922322dd7f62cc395c527061/packages/react/src/components/OverflowMenu/OverflowMenu.tsx#L585-L593

**Comment by divya-281** [comment]
@tay1orjones @guidari I added a test to check if the menu closes when pressing the esc key

**Comment by divya-281** [comment]
@tay1orjones @guidari Thanks for the suggestions, I checked locally and the coverage was 80%.

**Comment by divya-281** [comment]
@tay1orjones  @guidari Thanks for the suggestions, I was able to get the coverage upto 80%.

**Comment by guidari** [comment]
Thank you @divya-281 
One question, is there any reason why we can't test the `left` `right` in th switch case? I see `top` `bottom` are being tested.
Screenshot 2024-11-12 at 09 50 32

**Comment by divya-281** [comment]
@guidari  
const triggerButtonPositionProps = {
  [DIRECTION_TOP]: 'bottom',
  [DIRECTION_BOTTOM]: 'top',
};
 I only see bottom and top here, Can you let me know when the case left is triggered? Am I missing something?

**Comment by guidari** [comment]
Hey @divya-281 Interesting catch! I was taking a look in the code and seems like the `left` and `right` are never called.
The different types of direction are only called in the FeatureFlag OverflowMenu with `menuAlignment`
I guess we can comment or get rid of this line of code.

**Comment by divya-281** [comment]
@guidari  I've commented those lines

File: packages/react/src/components/OverflowMenu/OverflowMenu.tsx [hunk 1]
```diff
@@ -81,15 +82,16 @@ export const getMenuOffset = (menuBody, direction, trigger, flip) => {
...
+    // case left and right is not called
+    // case 'left':
+    // case 'right': {
+    //   // TODO: Ensure `trigger` is there for `<OverflowMenu open>`
+    //   const triggerHeight = !trigger ? 0 : trigger.offsetHeight;
+    //   return {
+    //     left: 0,
+    //     top: (!flip ? 1 : -1) * (menuHeight / 2 - triggerHeight / 2),
+    //   };
+    // }
```
Comment (tay1orjones) [inline packages/react/src/components/OverflowMenu/OverflowMenu.tsx]:
```suggestion
```

**Review by tay1orjones — APPROVED** [review]
LGTM, thanks!

**Review by guidari — APPROVED** [review]
Thank you @divya-281

**Pull request merged**